### PR TITLE
SNHUT - 2705 Integrate get users api for group details

### PR DESCRIPTION
**Pull request by lekshmic-sadaindia**
### Description

- API Integration ticket for listing users of a group

#### Screenshots
Screen Shot 2022-12-28 at 1 45 14 PM

Update:
02 nov 2023

working as expected
Screen Shot 2023-02-02 at 5 37 40 PM

### Other Details

- [JIRA Ticket](https://collectiveshift.atlassian.net/browse/SNHUT-2705)

#### Checklist

- [ ] **Assumptions of User Story met (including acceptance criteria)**
- [ ] **All new necessary Unit tests were CREATED** to promote code coverage
- [ ] **All required unit tests are PASSING**
- [ ] The proposed functionality was tested manually locally, and it's working fine.
- [ ] PR was properly reviewed by a team member (at least 1 approval required to merge!)
- [ ] Unnecessary console.logs AND/OR comments were removed
- [ ] Feature ok-ed by UX designer (If needed - not for back end components) and Product Owner
- [ ] Ensure configuration and/or build changes documented

### Pending Task
Data returning from api is not accurate, hence testing is blocked

### CODE REVIEW

**Comment by lekshmic-sadaindia** [comment]
Moving to draft since dev testing is still pending and have dependency on gp-core.

**Review by shafi-sadaindia — APPROVED** [review]
LGTM

File: apps/toboggan-api/src/app/modules/groups/groups.service.ts [hunk 1]
```diff
@@ -56,8 +59,8 @@ export class GroupsService {
     return this.httpService.post('/group', newGroup);
   }
 
-  updateGroup(uuid: string, updatedGroup: Partial<CreateGroupDTO>) {
-    return this.httpService.put(`/group/${uuid}`, updatedGroup);
+  updateGroup(uuid: string, user_ids: any) {
+    return this.httpService.put(`/group/${uuid}`, user_ids);
```
Comment (sada-ajith) [inline apps/toboggan-api/src/app/modules/groups/groups.service.ts:63]: are we passing user_ids for updating group. ?

File: apps/toboggan-app/src/app/group/components/add-users/add-users.component.spec.ts [hunk 2]
```diff
@@ -113,19 +113,17 @@ describe('AddUsersComponent', () => {
   });
 
   it('should call update call updategroup with valid user ', () => {
-    const approveGroupSpy = jest.spyOn(groupService, 'updateGroup');
+    const approveGroupSpy = jest.spyOn(groupService, 'addUsersToGroup');
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/group/components/add-users/add-users.component.spec.ts:116]: shouldn't it be calling updateGroup ?

File: apps/toboggan-app/src/app/group/components/list-users/list-users.component.ts [hunk 3]
```diff
@@ -63,13 +64,14 @@ export class ListUsersComponent implements OnInit, OnDestroy {
   };
 
   @Input() group: IGroup = {} as IGroup;
-
+  id: any;
```
Comment (sada-ajith) [inline apps/toboggan-app/src/app/group/components/list-users/list-users.component.ts:67]: Please rename variable according to context

**Review by sada-ajith — CHANGES_REQUESTED** [review]
PLease check my comments

File: apps/toboggan-api/src/app/modules/groups/groups.service.ts (hunk 1 above)
Comment (lekshmic-sadaindia) [reply apps/toboggan-api/src/app/modules/groups/groups.service.ts:63]: Good catch!

File: apps/toboggan-app/src/app/group/components/add-users/add-users.component.spec.ts (hunk 2 above)
Comment (lekshmic-sadaindia) [reply apps/toboggan-app/src/app/group/components/add-users/add-users.component.spec.ts:116]: No that logic is changed. we've new api endpoints to add and remove users from group

    component:      await this.groupService.**addUsersToGroup**(body, this.group.uuid);
[https://snhu-glidepath-dev-api.cloudpssolutions.com/docs/user-management/api/v1/redoc#tag/Group/operation/add_user_to_group_group__uuid__add_users_post](    https://snhu-glidepath-dev-api.cloudpssolutions.com/docs/user-management/api/v1/redoc#tag/Group/operation/add_user_to_group_group__uuid__add_users_post)

**Review by shafi-sadaindia — APPROVED** [review]
LGTM

**Comment by lekshmic-sadaindia** [comment]
> PLease check my comments

Addressed review comments

File: apps/toboggan-app/src/app/group/components/list-users/list-users.component.ts (hunk 3 above)
Comment (lekshmic-sadaindia) [reply apps/toboggan-app/src/app/group/components/list-users/list-users.component.ts]: Variable renamed.

**Review by sada-ajith — APPROVED** [review]

**Pull request merged**